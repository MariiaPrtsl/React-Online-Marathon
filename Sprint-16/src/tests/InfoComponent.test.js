import mockAxios from "../_mocks_/axios"
import renderer from 'react-test-renderer'; 
import getGitHubUser from "../services/DataService";

describe("Test comnonent Info", () => {
  test('should return gitHub user information', async () => {
    const userInfo = {data:{name:'yurkovskiy', url: 'https://api.github.com/users/yurkovskiy'}}
    
    mockAxios(userInfo.data.name);
    const resp = await getGitHubUser('yurkovskiy');
    expect(userInfo.data.url).toMatch(resp.data.url);
  });

  test('header renders correctly', () => {
    const wrapper = renderer.create(<h3>GitHub User Info</h3>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
  test('first item renders correctly', () => {
    const wrapper = renderer.create(<li>"login: yurkovskiy"</li>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
   
});
   
