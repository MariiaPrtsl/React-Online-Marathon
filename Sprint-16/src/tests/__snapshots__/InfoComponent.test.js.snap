// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test comnonent Info first item renders correctly 1`] = `
<li>
  "login: yurkovskiy"
</li>
`;

exports[`Test comnonent Info header renders correctly 1`] = `
<h3>
  GitHub User Info
</h3>
`;
